service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderController.java

```diff
@@ -0,0 +1,123 @@
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validateGroupService.validateGroup(caseData, C21CaseOrderGroup.class))
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        String index = (caseData.getC21OrderBundle() != null)
+            ? Integer.toString(caseData.getC21OrderBundle().size() + 1) : "1";
+
+
+        Document c21Document = getDocument(
+            authorization,
+            userId,
+            index,
+            createC21OrderService.getC21OrderTemplateData(caseData));
+
+        data.put("temporaryC21Order", addDocumentToC21Order(caseData, c21Document));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(data).build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
+        @RequestBody CallbackRequest callbackRequest) {
+        Map<String, Object> data = callbackRequest.getCaseDetails().getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        data.put("c21OrderBundle", createC21OrderService.addToC21OrderBundle(
+            caseData.getTemporaryC21Order(), caseData.getJudgeAndLegalAdvisor(), caseData.getC21OrderBundle()));
+        data.remove("temporaryC21Order");
+        data.remove("judgeAndLegalAdvisor");
+
+        return AboutToStartOrSubmitCallbackResponse.builder().data(data).build();
+    }
+
+    private C21Order addDocumentToC21Order(CaseData caseData, Document document) {
+        return caseData.getTemporaryC21Order().toBuilder()
+            .c21OrderDocument(DocumentReference.builder()
+                .url(document.links.self.href)
+                .binaryUrl(document.links.binary.href)
+                .filename(document.originalDocumentName)
+                .build())
+            .orderTitle(defaultIfBlank(caseData.getTemporaryC21Order().getOrderTitle(), "Order"))
+            .build();
+    }
+
+    private Document getDocument(@RequestHeader("authorization") String authorization,
+                                 @RequestHeader("user-id") String userId,
+                                 String index,
+                                 Map<String, Object> templateData) {
+        DocmosisDocument document = docmosisService.generateDocmosisDocument(templateData, C21);
+        return uploadDocumentService.uploadPDF(userId, authorization, document.getBytes(),
+            C21.getDocumentTitle() + "_" + index + ".pdf");
```

Comment:
Why the long face "_"?

Reply:
As previous comment, wanted to have name "C21_Order_1.pdf"... "C21_Order_2.pdf"... etc